src/integration/wrapper.go

```diff
@@ -0,0 +1,123 @@
+package integration
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	sdk "github.com/newrelic/infra-integrations-sdk/integration"
+	"github.com/newrelic/nri-kubernetes/v3/internal/config"
+	"github.com/newrelic/nri-kubernetes/v3/internal/logutil"
+	"github.com/newrelic/nri-kubernetes/v3/internal/storer"
+	"github.com/newrelic/nri-kubernetes/v3/src/integration/prober"
+	"github.com/newrelic/nri-kubernetes/v3/src/integration/sink"
+	"github.com/sethgrid/pester"
+	log "github.com/sirupsen/logrus"
+)
+
+const defaultProbeTimeout = 90 * time.Second
+const defaultProbeBackoff = 5 * time.Second
+
+const agentReadyPath = "/v1/data/ready"
+
+// Wrapper is a wrapper on top of the SDK integration.
+type Wrapper struct {
+	sdkIntegration *sdk.Integration
+	logger         *log.Logger
+	metadata       Metadata
+	probeTimeout   time.Duration
+	probeBackoff   time.Duration
+}
+
+// OptionFunc is an option func for the Wrapper.
+type OptionFunc func(i *Wrapper) error
+
+func WithLogger(logger *log.Logger) OptionFunc {
+	return func(i *Wrapper) error {
+		i.logger = logger
+		return nil
+	}
+}
+
+// WithProbeTimeout configures the integration wrapper to wait at most timeout for the HTTP endpoint to respond.
+func WithProbeTimeout(timeout time.Duration) OptionFunc {
+	return func(i *Wrapper) error {
+		i.probeTimeout = timeout
+		return nil
+	}
+}
+
+// WithProbeBackoff configures the time the internal prober waits between HTTP endpoint checks.
+func WithProbeBackoff(backoff time.Duration) OptionFunc {
+	return func(i *Wrapper) error {
+		i.probeBackoff = backoff
+		return nil
+	}
+}
```

Comment:
I am not a fan of adding unused code just in case it is going to be useful in the future

Reply:
When I added it I planned to add the relevant config options, but in the end I never did.

Do you think there is some value in adding these to the config?

Reply:
mmm not really, at least I cannot imagine a scenario in which a customer want to set those up

Reply:
Done, got rid of them :)